src/components/ImageWithSizeCalculation.js

```diff
@@ -39,71 +57,33 @@ class ImageWithSizeCalculation extends PureComponent {
 
         this.imageLoadingStart = this.imageLoadingStart.bind(this);
         this.imageLoadingEnd = this.imageLoadingEnd.bind(this);
+        this.onError = this.onError.bind(this);
+        this.imageLoadedSuccessfuly = this.imageLoadedSuccessfuly.bind(this);
     }
 
-    componentDidMount() {
-        this.calculateImageSize();
+    onError() {
+        Log.hmmm('Unable to fetch image to calculate size', {url: this.props.url});
     }
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.url === this.props.url) {
-            return;
-        }
-
-        this.calculateImageSize();
-    }
-
-    componentWillUnmount() {
-        if (!this.getImageSizePromise) {
-            return;
-        }
-
-        this.getImageSizePromise.cancel();
+    imageLoadingStart() {
+        this.setState({isLoading: true});
     }
 
-    /**
-     * @param {String} url
-     * @returns {Promise}
-     */
-    getImageSize(url) {
-        return new Promise((resolve, reject) => {
-            Image.getSize(url, (width, height) => {
-                resolve({width, height});
-            }, (error) => {
-                reject(error);
-            });
-        });
+    imageLoadingEnd() {
+        this.setState({isLoading: false});
     }
 
-    calculateImageSize() {
-        if (!this.props.url) {
+    imageLoadedSuccessfuly(event) {
+        if (!lodashGet(event, 'nativeEvent.width', false) || !lodashGet(event, 'nativeEvent.height', false)) {
+            // Image didn't load properly
             return;
         }
 
-        this.getImageSizePromise = makeCancellablePromise(this.getImageSize(this.props.url));
-        this.getImageSizePromise.promise
-            .then(({width, height}) => {
-                if (!width || !height) {
-                    // Image didn't load properly
-                    return;
-                }
-
-                this.props.onMeasure({width, height});
-            })
-            .catch((error) => {
-                Log.hmmm('Unable to fetch image to calculate size', {error, url: this.props.url});
-            });
-    }
-
-    imageLoadingStart() {
-        this.setState({isLoading: true});
-    }
-
-    imageLoadingEnd() {
-        this.setState({isLoading: false});
+        this.props.onMeasure({width: event.nativeEvent.width, height: event.nativeEvent.height});
     }
 
     render() {
+        const headers = this.props.isAuthTokenRequired ? chatAttachmentTokenHeaders() : undefined;
```

Comment:
IMO it might be better to forward `isAuthTokenRequired` to the `Image` and let the `Image` run this logic instead of having it repeat for any subtype of Image we have